common/src/main/kotlin/de/hpi/cloud/common/grpc/ServiceImpl.kt

```diff
@@ -1,20 +1,24 @@
-package de.hpi.cloud.common.utils.grpc
+package de.hpi.cloud.common.grpc
 
+import de.hpi.cloud.common.Context
+import de.hpi.cloud.common.Service
 import io.grpc.BindableService
 import io.grpc.Status
 import io.grpc.StatusRuntimeException
 import io.grpc.stub.StreamObserver
 
+@Suppress("unused") // Receiver is used for scoping
 fun <Req : Any, Res : Any> BindableService.catchErrors(
     request: Req?,
     responseObserver: StreamObserver<Res>?,
-    lambda: (Req, StreamObserver<Res>) -> Unit
+    body: Context.(Req, StreamObserver<Res>) -> Unit
 ) {
     try {
         request ?: Status.INVALID_ARGUMENT.throwException("request is null")
         responseObserver ?: Status.INVALID_ARGUMENT.throwException("responseObserver is null")
 
-        lambda(request, responseObserver)
+        val context = Service.contextForRequest(request) ?: error("Metadata not found")
```

Comment:
is this error relevant for the user? like BAD_REQUEST?

Reply:
`Context` not found is an internal error: The call was not intercepted by our registered `ServerInterceptor` or this interception did not register the call. This does not infer a client error